appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/ResourceValidator.java

```diff
@@ -0,0 +1,696 @@
+                jndiName = descriptor.getActivationConfigValue("destinationLookup");
+            }
+            // validateJNDIRefs(jndiName, ejb);
+        }
+        for (Object next : ejb.getResourceReferenceDescriptors()) {
+            accept((ResourceReferenceDescriptor) next, ejb);
+        }
+
+        for (Object next : ejb.getResourceEnvReferenceDescriptors()) {
+            accept((ResourceEnvReferenceDescriptor) next, ejb);
+        }
+
+        for (Object next : ejb.getMessageDestinationReferenceDescriptors()) {
+            accept((MessageDestinationReferenceDescriptor) next, ejb);
+        }
+
+        for (Object next : ejb.getEnvironmentProperties()) {
+            accept((EnvironmentProperty) next, ejb);
+        }
+
+        for (Object next : ejb.getEjbReferenceDescriptors()) {
+            accept((EjbReferenceDescriptor) next, ejb);
+        }
+    }
+
+    private void accept(BundleDescriptor bd) {
+        if (bd instanceof JndiNameEnvironment) {
+            JndiNameEnvironment nameEnvironment = (JndiNameEnvironment) bd;
+            for (Object next : nameEnvironment.getResourceReferenceDescriptors()) {
+                accept((ResourceReferenceDescriptor) next, nameEnvironment);
+            }
+
+            for (Object next : nameEnvironment.getResourceEnvReferenceDescriptors()) {
+                accept((ResourceEnvReferenceDescriptor) next, nameEnvironment);
+            }
+
+            for (Object next : nameEnvironment.getMessageDestinationReferenceDescriptors()) {
+                accept((MessageDestinationReferenceDescriptor) next, nameEnvironment);
+            }
+
+            for (Object next : bd.getMessageDestinations()) {
+                accept((MessageDestinationDescriptor) next, nameEnvironment);
+            }
+
+            for (Object next : nameEnvironment.getEnvironmentProperties()) {
+                accept((EnvironmentProperty) next, nameEnvironment);
+            }
+
+            for (Object next : nameEnvironment.getEjbReferenceDescriptors()) {
+                accept((EjbReferenceDescriptor) next, nameEnvironment);
+            }
+
+            for (PersistenceUnitsDescriptor pus : bd.getExtensionsDescriptors(PersistenceUnitsDescriptor.class)) {
+                for (PersistenceUnitDescriptor pu : pus.getPersistenceUnitDescriptors()) {
+                    accept(pu, nameEnvironment);
+                }
+            }
+
+            for (ManagedBeanDescriptor mbd: bd.getManagedBeans()) {
+                accept(mbd, nameEnvironment);
+            }
+        }
+    }
+
+    /**
+     * Validate resources stored in ResourceRefDescriptor.
+     */
+    private void accept(ResourceReferenceDescriptor resRef, JndiNameEnvironment env) {
+        String jndiName = resRef.getJndiName();
+
+        if (resRef.isWebServiceContext())
+            return;
+
+        if (resRef.isURLResource()) {
+            if (jndiName != null && !(jndiName.startsWith(ResourceConstants.JAVA_SCOPE_PREFIX))) {
```

Comment:
It can be a custom resource also. Something defined using the create-custom-resource command. Refer to one of the devtests I have written for an example.